src/site/content/en/blog/async-functions/index.md

```diff
@@ -0,0 +1,392 @@
+---
+title: "Async functions: making promises friendly"
+subhead: >
+  Async functions allow you to write promise-based code as if it were synchronous.
+description: >
+  Async functions allow you to write promise-based code as if it were synchronous.
+date: 2016-10-20
+updated: 2021-01-18
+tags:
+  - javascript
+authors:
+  - jakearchibald
+feedback:
+  - api
+---
+
+Async functions were enabled by default in Chrome 55 and they're quite frankly
+marvelous. They allow you to write promise-based code as if it were synchronous,
+but without blocking the main thread. They make your asynchronous code less
+"clever" and more readable.
+
+Async functions work like this:
+
+```js
+async function myFirstAsyncFunction() {
+  try {
+    const fulfilledValue = await promise;
+  }
+  catch (rejectedValue) {
+    // …
+  }
+}
+```
+
+If you use the `async` keyword before a function definition, you can then use
+`await` within the function. When you `await` a promise, the function is paused
+in a non-blocking way until the promise settles. If the promise fulfills, you
+get the value back. If the promise rejects, the rejected value is thrown.
+
+{% Aside %}
+If you're unfamiliar with promises, check out [our
+promises guide](/promises).
+{% endAside %}
+
+## Example: logging a fetch
+
+Say you want to fetch a URL and log the response as text. Here's how it looks
+using promises:
+
+```js
+function logFetch(url) {
+  return fetch(url)
+    .then(response => response.text())
+    .then(text => {
+      console.log(text);
+    }).catch(err => {
+      console.error('fetch failed', err);
+    });
+}
+```
+
+And here's the same thing using async functions:
+
+```js
+async function logFetch(url) {
+  try {
+    const response = await fetch(url);
+    console.log(await response.text());
+  }
+  catch (err) {
+    console.log('fetch failed', err);
+  }
+}
+```
+
+It's the same number of lines, but all the callbacks are gone. This makes it way
+easier to read, especially for those less familiar with promises.
+
+{% Aside %}
+Anything you `await` is passed through `Promise.resolve()`, so you can
+safely `await` non-native promises.
```

Comment:
@jpmedley please rephrase to avoid "native"

Reply:
@jakearchibald I don't know what non-native promises are.

Reply:
Added suggestion below